GliaWidgets/Sources/QuickLookBased/QuickLookBased.Implementation.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+import UIKit
+import QuickLook
+
+extension QuickLookBased.ThumbnailGenerator {
+    static func create(uiScreen: UIKitBased.UIScreen) -> Self {
+        if #available(iOS 13.0, *) {
+            let generator = QLThumbnailGenerator.shared
+            return .init(
+                generateBestRepresentation: { size, url, completion in
+                    let scale = uiScreen.scale()
+                    let newSize = CGSize(width: size.width * scale, height: size.height * scale)
+                    let request = QLThumbnailGenerator.Request(
+                        fileAt: url.standardizedFileURL,
+                        size: newSize,
+                        scale: scale,
+                        representationTypes: .thumbnail
+                    )
+                    generator.generateBestRepresentation(for: request) { thumbnail, error in
+                        if let image = thumbnail?.uiImage {
+                            let cropped = image.resized(to: size)
+                            completion(cropped)
+                        } else if error != nil {
+                            // TODO: need to handle error state
```

Comment:
Is there a task for this?

Reply:
I would raise discussion about handling this error state. Because it might require some design work. Will do this right now

Reply:
https://salemove.slack.com/archives/C01F9K88SQZ/p1684403981765899